build-cran-package.sh

```diff
@@ -130,7 +130,7 @@ cd "${TEMP_R_DIR}"
     # Rtools35 (used with R 3.6 on Windows) doesn't support C++17
     LGB_CXX_STD="C++17"
     using_windows_and_r3=$(
-        Rscript -e 'cat(.Platform$OS.type == "windows" && R.version[["major"]] < 4)'
+        Rscript -e "cat(.Platform$OS.type == \"windows\" && R.version[[\"major\"]] < 4)"
```

Comment:
Did `shellcheck` suggest this? What was the warning/error code and text?

If yes, could you please just ignore that check for this line with a comment similar to https://github.com/microsoft/LightGBM/blob/41ba9e8f00c89d72e5cb71c964722ce1ed4d8445/docs/build-docs.sh#L23

This is intentionally using single quotes so that inner `$` is treated as a literal `$` and not used for string interpolation by the shell. I strongly suspect this change is going to break this line of code otherwise.

Reply:
Yeah I see the change has broken some windows CIs, will revert